src/nrtk_explorer/app/transforms.py

```diff
@@ -308,15 +308,16 @@ def _update_images(self, selected_ids):
     async def _set_source_images(self, selected_ids):
         # We need to yield twice for the self.state.loading_images=True to
         # commit to the trame state to show a spinner
-        await asyncio.sleep(0)
-        await asyncio.sleep(0)
+        # await asyncio.sleep(0)
+        # await asyncio.sleep(0)
         self._update_images(selected_ids)
         with self.state:
             self.state.loading_images = False
 
     def set_source_images(self, selected_ids):
         if len(selected_ids):
             self.state.loading_images = True
+            self.state.flush()
```

Comment:
@jourdain To kick off the client side spinner when selecting images we have these two beauties:
```
await asyncio.sleep(0)
await asyncio.sleep(0)
```
🤮

`state.flush()` does not seem to push out the `state.loading_images = True` change.

Is there a better way to update the `state` client side before kicking off some long running Python?

Reply:
you do need to give room for the network to kick in. So you need something like `await asyncio.sleep(0.1)`.
Not sure it is better than 2x0 sleep?

Reply:
another approach could be to add some helper like

```python
await trame.network.next()
```

Reply:
```
await asyncio.sleep(0)
await asyncio.sleep(0)
```

Sleep 0 is a shortcut to yield to the other thread/task in Trame to execute. We found either that this had to be done twice for it to work, or instead we can wait some amount of time (we tested 1s) for the changes to be pushed. The first option was preferred since it was faster (no time sleeping).

Reply:
Updated the code.  I make a context manager to update state inside of async funcs:
```py
    async def _update_images(self):
        async with SetStateAsync(self.state) as state:
            state.loading_images = True
```

The context manager is using `await asyncio.sleep(0.1)` cuz double `asyncio.sleep(0)` did not cut it in `TransformsApp._update_images` 😬

So some sort of guaranteed network flush trick seems essential.

Reply:
Anything that gets the job done is best, I like that you factored this out in its own class with dunder methods. Before we move to the `0.1`, what is it better. We considered this back that the time and the 0.1 was flaky at times and it felt fragile since sometimes it needed a second yield. note that in python async `await asyncio.sleep(0)` is an specific yield instruction.

Reply:
Interesting.  I'm finding both are flakey
```
await asyncio.sleep(0) # reliably does not work as I start trying jam more 
await asyncio.sleep(0)
```
and 
```
await asyncio.sleep(.1) # does not (always?) work if I select more than 2 images
```
So hoping something turns up or else async pushing to state is borked.